src/lib/party.ts

```diff
@@ -516,8 +577,105 @@ class Party {
     return new (Party as any)(body, authorization, party.isSelf);
   }
 
+  token(tokenId?: Field) {
+    let thisParty = this;
+    let customToken = new Token({
+      tokenOwner: thisParty.body.publicKey,
+      parentTokenId: tokenId,
+    });
+    return {
+      id: customToken.id,
+      parentTokenId: customToken.parentTokenId,
+      tokenOwner: customToken.tokenOwner,
+
+      mint({ address, amount }: MintOrBurnParams) {
+        let receiverParty = Party.createUnsigned(address, {
+          caller: this.id,
+          tokenId: this.id,
+          callDepth: 1,
+          useFullCommitment: Bool(true),
+        });
+
+        // Add the amount to mint to the receiver's account
+        receiverParty.body.balanceChange =
+          receiverParty.body.balanceChange.add(amount);
+      },
+
+      burn({ address, amount }: MintOrBurnParams) {
+        let receiverParty = Party.createUnsigned(address, {
+          caller: this.id,
+          tokenId: this.id,
+          callDepth: 1,
+          useFullCommitment: Bool(true),
+        });
+
+        // Sub the amount to burn from the receiver's account
+        receiverParty.body.balanceChange =
+          receiverParty.body.balanceChange.sub(amount);
+
+        // Require signature from the receiver account being deducted
+        receiverParty.authorization = {
+          kind: 'lazy-signature',
+        };
+      },
+
+      send({ from, to, amount }: SendParams) {
+        // Check if the sender party is the current zkApp address
+        if (from === thisParty.publicKey) {
```

Comment:
hm, just using `from === thisParty.publicKey` to compare public keys seems wrong and clever at the same time 😆

Reply:
That was a bad call on my part :sweat_smile: Ended up removing this check due to the comment you mentioned about the conditions being the same thing.